### PR TITLE
Kamil

### PR DESCRIPTION
**Pull request by Kamil-Jankowski**
Code updated to include changes created on Hubert branch.

### CODE REVIEW

File: src/app/game/game.component.html [hunk 1]
```diff
@@ -7,7 +7,18 @@
...
         </div>
-        <app-ship-map></app-ship-map>
+        <mat-grid-list cols="10" rowHeight="1:1">
+          <div *ngFor="let square of shipMap">
+            <mat-grid-tile>
+              <button mat-mini-fab color="" 
+                [ngClass]="{'button-miss': square.status==2,
+                 'button-hit': square.status==1,
+                  'button-ship': square.status==3,
+                  'button-blocked': square.status==0}">
```
Comment (MJ211) [inline src/app/game/game.component.html:17]: magic numbers

File: src/app/game/game.component.html [hunk 2]
```diff
@@ -19,8 +30,20 @@
...
-      </div>
+        <mat-grid-list cols="10" rowHeight="1:1">
+          <div *ngFor="let square of shootMap">
+            <mat-grid-tile>
+              <button mat-mini-fab color="" 
+              [ngClass]="{'button-miss' : square.status==2,
+               'button-hit': square.status==1,
+                'button' : square.status==4,
+                 'button-blocked' : square.status==0}" 
+                  (click)="status = changeStatus(square.id)">
```
Comment (MJ211) [inline src/app/game/game.component.html:41]: .

File: src/app/game/game.component.ts [hunk 3]
```diff
@@ -17,7 +26,25 @@ import { TranslateService } from '@ngx-translate/core';
...
+  private isFirstTurn : boolean = true;
+
+
+  shipMap: Square[] = [];
+  shootMap: Square[] = [];
+
+  hitMessage: string;
+  missMessage: string;
+  turnMessage: string;
+  enemyTurnMessage: string;
```
Comment (MJ211) [inline src/app/game/game.component.ts]: Try to set variables private as many as you can, and sort them

File: src/app/game/game.component.ts [hunk 4]
```diff
@@ -27,24 +54,168 @@ export class GameComponent implements OnInit {
    */
   constructor(
     private route: ActivatedRoute,
+    private router: Router,
     private playerService: PlayerService,
-    public messageService: MessageService,
+    public gameService: GameService,
+    private notificationService: NotificationService,
     public translate: TranslateService) {
```
Comment (MJ211) [inline src/app/game/game.component.ts]: .

File: src/app/game/game.component.ts [hunk 5]
```diff
@@ -27,24 +54,168 @@ export class GameComponent implements OnInit {
...
    * Assigns player based on the url route
    */
-  getPlayer(): void {
-    const id = +this.route.snapshot.paramMap.get('id');
-    this.playerService.getPlayer(id)
-        .subscribe(player => this.player = player);
+  private getPlayer(): void{
+    const name :string = this.route.snapshot.paramMap.get('name');
+    this.player = {name};
+    console.log(`player name is ${name}`);
```
Comment (MJ211) [inline src/app/game/game.component.ts]: Is this needed?

File: src/app/game/game.component.ts [hunk 6]
```diff
@@ -27,24 +54,168 @@ export class GameComponent implements OnInit {
...
+     this.shootMap[key] = {id : parseInt(key),
+        status : shootMap[key] === ShootMapCellStatus.SHOOT_MAP_MISS ? 2 :1};
+     });
+ }
+
+ private mapShipMapToArray(shipMap : Map<number, ShipMapCellStatus>) : void{
+   Object.keys(shipMap).forEach( key => {
+     this.shipMap[key] = {id : parseInt(key),
+        status : shipMap[key] === ShipMapCellStatus.SHIP_MAP_MISS ? 2 :
+        shipMap[key] === ShipMapCellStatus.SHIP_MAP_SHIP ? 3 :1};
```
Comment (MJ211) [inline src/app/game/game.component.ts]: Magic numbers

File: src/app/game/game.component.ts [hunk 7]
```diff
@@ -27,24 +54,168 @@ export class GameComponent implements OnInit {
...
+          }
+        } else if(cellStatus === ShootMapCellStatus.SHOOT_MAP_MISS){
+          currentButton.status = ButtonStatus.MISS;
+          this.showMessage(this.enemyTurnMessage, NotificationType.info);
+          this.showMessage(this.missMessage, NotificationType.error);
+          this.playerTurn = this.opponent;
+          this.blockAllShotSquares();
+        }
+        this.shootMap[id] = currentButton;
+        console.log("After: ", this.shootMap[id].status);
```
Comment (MJ211) [inline src/app/game/game.component.ts]: Logs

File: src/app/landing/landing.component.ts [hunk 8]
```diff
@@ -14,18 +15,24 @@ import { NotificationType } from '../notification.message';
...
   ngOnInit(): void {
-    this.translate
+    let type :string = this.route.snapshot.paramMap.get('type');
+    if(type === "win"){
+      this.translate
         .get('TOAST.WIN')
         .subscribe((msg: string) => {this.notificationService.print(msg, NotificationType.warning)});
-    this.translate
+    }
+    else if(type == "loose"){
```
Comment (MJ211) [inline src/app/landing/landing.component.ts]: ===

**Review by MJ211 — COMMENTED** [review]
Done

File: src/app/game/game.component.html [hunk 9]
```diff
@@ -11,7 +11,10 @@
           <div *ngFor="let square of shipMap">
             <mat-grid-tile>
               <button mat-mini-fab color="" 
-                [ngClass]="{'button-miss': square.status==2, 'button-hit': square.status==1, 'button-ship': square.status==3}">
+                [ngClass]="{'button-miss': square.status==2,
```
Comment (Kamil-Jankowski) [inline src/app/game/game.component.html:14]: Change needed to reflect introduced enum

File: src/app/game/game.component.html [hunk 10]
```diff
@@ -30,8 +33,12 @@
         <mat-grid-list cols="10" rowHeight="1:1">
           <div *ngFor="let square of shootMap">
             <mat-grid-tile>
-              <button mat-mini-fab color="" [ngClass]="{'button-miss' : square.status==2, 'button-hit': square.status==1, 'button-blocked' : square.status==4}" 
-                                              (click)="status = changeStatus(square.id)">
+              <button mat-mini-fab color=""
```
Comment (Kamil-Jankowski) [inline src/app/game/game.component.html:36]: Change needed to reflect introduced enum

File: src/app/game/game.component.html (hunk 1 above)
Comment (hubert-milczarek-EPAM) [reply src/app/game/game.component.html:17]: separate task is created for that issue

File: src/app/game/game.component.ts (hunk 3 above)
Comment (MJ211) [reply src/app/game/game.component.ts]: I meant: sort by access specifier

File: src/app/game/game.component.html (hunk 9 above)
Comment (hubert-milczarek-EPAM) [reply src/app/game/game.component.html:14]: separate task created for that issue

File: src/app/game/game.component.html (hunk 10 above)
Comment (hubert-milczarek-EPAM) [reply src/app/game/game.component.html:36]: separated task created for that issue

File: src/app/game/game.component.ts [hunk 11]
```diff
@@ -26,25 +52,164 @@ export class GameComponent implements OnInit {
...
+          this.router.navigate(['/landing/win']);
+        }
+        else if(currentGameStatus.playerNameWhoMoves === this.player.name && currentGameStatus.playerLooser){
+          this.router.navigate(['/landing/loose']);
+        }
+        this.playerTurn = {name : currentGameStatus.playerNameWhoMoves};
+    } );
+  }
+
+  ngOnDestroy(): void{
```
Comment (Kamil-Jankowski) [inline src/app/game/game.component.ts:108]: After the player surrenders, it is removed from the db. When you try to access his game screen it again redirects to landing/loose.
However, the winner is redirected at first, but it is possible to access his game screen after that.

**Pull request merged**